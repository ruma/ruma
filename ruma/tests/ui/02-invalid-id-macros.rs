fn main() {
    let _ = ruma::device_key_id!("ed2519:JLAFKJWSCS");
    let _ = ruma::event_id!("39hvsi03hlne:example.com");
    let _ = ruma::event_id!("acR1l0raoZnm60CBwAVgqbZqoO/mYU81xysh1u7XcJk");
    let _ = ruma::room_alias_id!("alias:server.tld");
    let _ = ruma::room_id!("1234567890:matrix.org");
    let _ = ruma::room_version_id!("");
    let _ = ruma::server_key_id!("ed219:Abc_1");
    let _ = ruma::server_name!("");
    let _ = ruma::user_id!("user:ruma.io");
}
