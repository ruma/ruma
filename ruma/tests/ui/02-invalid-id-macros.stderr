error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:2:13
  |
2 |     let _ = ruma::device_key_id!("ed2519:JLAFKJWSCS");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid device key id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:3:13
  |
3 |     let _ = ruma::event_id!("39hvsi03hlne:example.com");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid event id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:4:13
  |
4 |     let _ = ruma::event_id!("acR1l0raoZnm60CBwAVgqbZqoO/mYU81xysh1u7XcJk");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid event id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:5:13
  |
5 |     let _ = ruma::room_alias_id!("alias:server.tld");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid room_alias_id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:6:13
  |
6 |     let _ = ruma::room_id!("1234567890:matrix.org");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid room_id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:7:13
  |
7 |     let _ = ruma::room_version_id!("");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid room_version_id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:8:13
  |
8 |     let _ = ruma::server_key_id!("ed219:Abc_1");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid server_key_id

error: proc macro panicked
 --> $DIR/02-invalid-id-macros.rs:9:13
  |
9 |     let _ = ruma::server_name!("");
  |             ^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid server_name

error: proc macro panicked
  --> $DIR/02-invalid-id-macros.rs:10:13
   |
10 |     let _ = ruma::user_id!("user:ruma.io");
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: Invalid user_id
